### PR TITLE
Move duckdb Parquet reader to separate folder

### PR DESCRIPTION
**Pull request by yingsu00**
This PR is to prepare for the upcoming native Parquet reader. It moves the duckdb Parquet reader into dwio/parquet/duckdb_reader. Users will be able to choose duckdb Parquet reader or the new native Parquet reader in the future.

### CODE REVIEW

File: velox/connectors/hive/FileHandle.h [hunk 1]
```diff
@@ -29,11 +29,11 @@
 #include <memory>
 #include <string>
 
+#include "dwio/common/InputStream.h"
```
Comment (mbasmanova) [inline velox/connectors/hive/FileHandle.h:32]: Why this change?

**Review by mbasmanova — COMMENTED** [review]
@yingsu00 This PR appears to have a lot of changes that are not in the parquet reader. Are these changes required? Can they be extracted into a separate PR?

**Comment by yingsu00** [comment]
@mbasmanova Thank you very much for reviewing the PR. The first two commits of this PR were actually [PR1754](https://github.com/facebookincubator/velox/pull/1754) and [PR1735](https://github.com/facebookincubator/velox/pull/1735). Those two PRs were part of the refactoring work of https://github.com/facebookincubator/velox/issues/1533. Sorry I didn't state clearly in the PR message. Now PR1754 was merged and I'm working on PR1735. I'll rebase this PR on top of latest 1735 tomorrow. It'll be nice if you could take a look at the last commit only. After 1735 is merged I'll rebase this one again.

File: velox/dwio/dwrf/test/utils/DataFiles.cpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include "velox/dwio/dwrf/test/utils/DataFiles.h"
+#include <iostream>
```
Comment (mbasmanova) [inline velox/dwio/dwrf/test/utils/DataFiles.cpp:17]: remove

File: velox/dwio/parquet/tests/ParquetReaderTestBase.h [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  template <typename T>
+  VectorPtr rangeVector(size_t size, T start) {
+    std::vector<T> vals(size);
+    for (size_t i = 0; i < size; ++i) {
+      vals[i] = start + static_cast<T>(i);
+    }
+    return vectorMaker_->flatVector(vals);
+  }
+
+  //  template <>
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/ParquetReaderTestBase.h:58]: Remove commented out code.

File: velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp [hunk 4]
```diff
@@ -25,18 +26,21 @@
...
+using namespace facebook::velox::parquet;
 
 class ParquetTableScanTest : public HiveConnectorTestBase {
  protected:
   using OperatorTestBase::assertQuery;
 
   void SetUp() override {
     HiveConnectorTestBase::SetUp();
-    parquet::registerParquetReaderFactory();
+    //    duckdb_reader::registerDuckDbParquetReaderFactory();
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp:37]: remove commented out code; ditto below

File: velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp [hunk 5]
```diff
@@ -107,7 +111,7 @@ class ParquetTableScanTest : public HiveConnectorTestBase {
 
   std::string getExampleFilePath(const std::string& fileName) {
     return facebook::velox::test::getDataFilePath(
-        "velox/dwio/parquet/tests", "examples/" + fileName);
+        "", "../examples/" + fileName);
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp:114]: use paths from the top of the repo

File: velox/dwio/parquet/tests/duckdb_reader/ParquetTpchTest.cpp [hunk 6]
```diff
@@ -15,21 +15,23 @@
  */
 
 #include <folly/init/Init.h>
+#include <vector>
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/duckdb_reader/ParquetTpchTest.cpp:18]: not needed

**Review by mbasmanova — COMMENTED** [review]
@yingsu00 3-rd commit looks good.

**Comment by yingsu00** [comment]
@mbasmanova All comments resolved and PR rebased. Your review is very appreciated! The  linux-benchmarks-basic test failure is waiting for PR1808 to be merged.

File: velox/dwio/parquet/tests/CMakeLists.txt [hunk 7]
```diff
@@ -18,50 +18,14 @@ include_directories(${CMAKE_BINARY_DIR})
...
-  velox_hive_connector
-  velox_aggregates
-  ${TEST_LINK_LIBS})
-
-add_executable(velox_dwio_parquet_tpch_test ParquetTpchTest.cpp)
-add_test(
-  NAME velox_dwio_parquet_tpch_test
-  COMMAND velox_dwio_parquet_tpch_test
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+message(STATUS "TEST_LINK_LIBS ${TEST_LINK_LIBS}")
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/CMakeLists.txt:29]: Not needed, right?

File: velox/dwio/parquet/tests/ParquetReaderTestBase.h [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include "velox/dwio/dwrf/test/utils/DataFiles.h"
+#include "velox/vector/tests/VectorMaker.h"
+
+using namespace ::testing;
```
Comment (mbasmanova) [inline velox/dwio/parquet/tests/ParquetReaderTestBase.h:24]: Is this needed?

**Review by mbasmanova — APPROVED** [review]
@yingsu00 Looks good. The build fix was merged. Please rebase.

File: velox/dwio/parquet/tests/ParquetReaderTestBase.h (hunk 8 above)
Comment (yingsu00) [reply velox/dwio/parquet/tests/ParquetReaderTestBase.h:24]: You're right, it's not needed. Thanks for the catch.

**Comment by yingsu00** [comment]
@mbasmanova Rebased and comments addressed. Thanks for reviewing!

**Comment by yingsu00** [comment]
@mbasmanova THank you for importing the PR. I see the two FB checks failed but couldn't see the details. Is there anything I can do to resolve them?

File: velox/benchmarks/tpch/TpchBenchmark.cpp [hunk 9]
```diff
@@ -22,7 +22,7 @@
 #include "velox/common/file/FileSystems.h"
 #include "velox/connectors/hive/HiveConnector.h"
 #include "velox/dwio/common/Options.h"
-#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/dwio/parquet/duckdb_reader/ParquetReader.h"
```
Comment (zzhao0) [inline velox/benchmarks/tpch/TpchBenchmark.cpp:25]: need to include "RegisterParquetReader.h" instead

File: velox/dwio/parquet/common/RegisterParquetReader.h [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+#pragma once
+
+#include "velox/dwio/parquet/duckdb_reader/ParquetReader.h"
+
+namespace facebook::velox::parquet {
+
+enum class ParquetReaderType { DUCKDB };
+
+void registerParquetReaderFactory(ParquetReaderType parquetReaderType) {
```
Comment (zzhao0) [inline velox/dwio/parquet/common/RegisterParquetReader.h:25]: let's move definition to .cpp file

File: velox/dwio/parquet/tests/ParquetReaderTestBase.h [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include "velox/dwio/dwrf/test/utils/DataFiles.h"
+#include "velox/vector/tests/VectorMaker.h"
+
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox;
```
Comment (zzhao0) [inline velox/dwio/parquet/tests/ParquetReaderTestBase.h:25]: let's remove these

**Review by zzhao0 — COMMENTED** [review]
Will have to clean up the fbcode side build files. Will import once you address the comments. Thanks

**Comment by yingsu00** [comment]
@zzhao0 Thank you very much for your careful review. I have addressed your comments, would you please take another look?

**Pull request closed without merging**